services/_shared/llm_client.py

```
@@ -758,7 +773,11 @@ def _call_backend(
         "response_format": {"type": "json_object"},
         **config.extra_body,
     }
-    headers = {"Authorization": f"Bearer {key}"}
+    if any(name.lower() == "authorization" for name in config.extra_headers):
+        raise _BackendConfigError(
+            f"{config.backend.value} extra_headers must not contain Authorization"
+        )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)